tests/server/testServerWithStorage.js

```diff
@@ -520,9 +639,148 @@ sjrk.storyTelling.server.testServerWithStorageDefs.testImageRetrieval = function
+        { block: { blockType: "audio" }, url: "testUrl", fieldToCheck: "mediaUrl", expected: "testUrl" }, // test audio block with set string
+        { block: { blockType: "video" }, url: "testUrl", fieldToCheck: "mediaUrl", expected: "testUrl" } // test video block with set string
+    ];
+
+    fluid.each(testCases, function (testCase) {
+        sjrk.storyTelling.server.setMediaBlockUrl(testCase.block, testCase.url);
+        var actual = testCase.block[testCase.fieldToCheck];
+
+        var message = "Media block URL is as expected: " + testCase.expected;
+        jqUnit.assertEquals(message, testCase.expected, actual);
+    });
+};
+
+sjrk.storyTelling.server.testServerWithStorageDefs.rotateImageFromExifTests = function () {
+    var testCases = [
+        { fileName: null, options: null, expectedResolution: false },
+        { fileName: "", options: null, expectedResolution: false },
+        { fileName: "", options: "", expectedResolution: false },
+        { fileName: "correctOrientation.jpg", options: null, expectedResolution: true },
+        { fileName: "incorrectOrientation.jpeg", options: null, expectedResolution: true },
+        { fileName: "test_gif.gif", options: null, expectedResolution: true },
+        { fileName: "logo_small_fluid_vertical.png", options: null, expectedResolution: true },
+        { fileName: "Leslie_s_Strut_Sting.mp3", options: null, expectedResolution: true },
+        { fileName: "shyguy_and_rootbeer.mp4", options: null, expectedResolution: true },
+        { fileName: "correctOrientation.jpg", options: { quality: 1 }, expectedResolution: true },
+        { fileName: "incorrectOrientation.jpeg", options: { quality: 1 }, expectedResolution: true },
+        { fileName: "test_gif.gif", options: { quality: 1 }, expectedResolution: true },
+        { fileName: "logo_small_fluid_vertical.png", options: { quality: 1 }, expectedResolution: true },
+        { fileName: "Leslie_s_Strut_Sting.mp3", options: { quality: 1 }, expectedResolution: true },
+        { fileName: "shyguy_and_rootbeer.mp4", options: { quality: 1 }, expectedResolution: true }
+    ];
+
+    fluid.each(testCases, function (testCase) {
+        var filePath = testCase.fileName;
+
+        // copy the file to the test uploads dir, if a filename was provided
+        if (filePath) {
+            var oldFilePath = "./tests/testData/" + testCase.fileName;
+            filePath = "./tests/server/uploads/" + uuidv1() + path.extname(testCase.fileName);
+            fs.copyFileSync(oldFilePath, filePath);
+        }
+
+        // call the function passing the new path and options where applicable
+        jqUnit.stop();
+        sjrk.storyTelling.server.rotateImageFromExif({ path: filePath }, testCase.options).then(function () {
+            jqUnit.assertEquals("Rotation call resolved as expected", testCase.expectedResolution, true);
```

Comment:
I think the tests for `rotateImageFromExif` should check images have been rotated to correct orientations. Why are assertions here and at line 719 checking resolutions rather than orientations?

Reply:
I have added assertions to the two cases where the image data is returned, i.e. the promise resolves but without "errors". The `correctOrientation` image will resolve but won't return image data since technically there was an error in the call to `jo.rotate`